CMakeLists.txt

```diff
@@ -0,0 +1,75 @@
+# Copyright (c) 2023-present The Bitcoin Core developers
+# Distributed under the MIT software license, see the accompanying
+# file COPYING or https://opensource.org/license/mit/.
+
+# Ubuntu 22.04 LTS Jammy Jellyfish, https://wiki.ubuntu.com/Releases, EOSS in June 2027:
+#  - CMake 3.22.1, https://packages.ubuntu.com/jammy/cmake
+#
+# Centos Stream 9, EOL in May 2027:
+#  - CMake 3.26.5, https://mirror.stream.centos.org/9-stream/AppStream/x86_64/os/Packages/
+cmake_minimum_required(VERSION 3.22)
+if(POLICY CMP0141)
+  # MSVC debug information format flags are selected by an abstraction.
+  # We want to use the CMAKE_MSVC_DEBUG_INFORMATION_FORMAT variable
+  # to select the MSVC debug information format.
+  cmake_policy(SET CMP0141 NEW)
+endif()
```

Comment:
> nit in [a2317e2](https://github.com/bitcoin/bitcoin/commit/a2317e27b7fb86df4e32cd1674c06e09cb808248): Maybe add a comment that this can be removed after `cmake_minimum_required(VERSION 3.25)`?

It seems redundant to me. Since policies are tied to specific versions,  it's generally understood that one should review every `if(CMAKE_VERSION...)` and `if(POLICY...)` statement in the code when updating the minimum required version.

> Alternatively, given that "The default installation of the latest version of MSVC 17.10.4 includes both CMake 3.28.3 and the vcpkg package manager", it may be easier to just require cmake >= 3.25 on MSVC and enforce the policy this way?

The build system becomes aware that it is building on MSVC only after the `project()` call, which makes such an implementation unfeasible given modern CMake requirements ([here](https://cmake.org/cmake/help/latest/command/cmake_minimum_required.html) and [here](https://cmake.org/cmake/help/latest/command/project.html)).

Besides, the current implementation is idiomatic for enforcing individual policies.

The other alternative is to specify `cmake_minimum_required(VERSION 3.22...3.25)`, but this has a drawback,  as the build system's behavior depends on the actual CMake version.

Reply:
Apparently, there is no need to enable the `CMP0141` policy, as we have not used the `CMAKE_MSVC_DEBUG_INFORMATION_FORMAT` variable since https://github.com/hebasto/bitcoin/pull/215.

Fixed in https://github.com/bitcoin/bitcoin/pull/30803.